Application/EdFi.Ods.Api/Helpers/AssemblyLoaderHelper.cs

```diff
@@ -191,7 +203,38 @@ static bool HasPlugin(Assembly assembly)
             }
         }
 
-        private class PluginAssemblyLoadingContext : AssemblyLoadContext
+        private static bool IsExtensionAssembly(AssemblyMetadata assemblyMetadata)
+        {
+            return assemblyMetadata.AssemblyModelType.EqualsIgnoreCase(PluginConventions.Extension);
+        }
+
+        private static bool IsProfileAssembly(AssemblyMetadata assemblyMetadata)
+        {
+            return assemblyMetadata.AssemblyModelType.EqualsIgnoreCase(PluginConventions.Profile);
+        }
+
+        private static AssemblyMetadata LoadAssemblyMetadata(Assembly assembly)
+        {
+            var resourceName = assembly.GetManifestResourceNames()
+                .SingleOrDefault(p => p.EndsWith(AssemblyMetadataSearchString));

```

Comment:
Now with the way this is written (to search for any embedded resources that end with "assemblyMetadata.json"), there is a chance (if the assembly author put multiple copies of the file at different locations) that we could end up with more than 1 matching embedded resource. In this case `SingleOfDefault` will throw an exception.

Let's change this to `FirstOrDefault`, so that we use the first one found (they _should_ only include 1, but you never know).